### DISCUSSION TITLE
COUNT ON FUNCTION AGREGATTE

### DISCUSSION THREAD

**Discussion by CACG1022** [opengisch/QField#4247] [Bugs]
The following is to report the following error since in previous versions the automatic counting of points worked correctly with versions 2.7.1 and 2.7.2 now with the new versions it stopped working.
The function listed the points located on the map sequentially, counting the points previously located for each polygon. The function used is the following: 
Image 1 shows the error and Image 2 shows that in previous versions it worked correctly, I don't understand why the update fixes something but damages others; Image 3 shows how the function is configured in the CONSECUTIVE field and that it worked correctly in previous versions

aggregate(layer:='AGREGAR',aggregate:='count',
expression:="CONSECUTIV",filter:="POLIGONO"=attribute( @parent, 'POLIGONO'))+1

IMAGEN 1
![help1](https://user-images.githubusercont
IMAGEN 2
![help2](https://user-images.githubusercontent.com/118937011/235368916-a8f300fb-8531-409a-8b5b-29d997a78441.png)
IMAGEN 3
![help3](https://user-images.githubusercontent.com/118937011/235369164-cd7c4c16-23b5-4d8f-8073-83abbb8e
[Archivos.zip](https://github.com/opengisch/QField/files/11361997/Archivos.zip)
4ac0.png)
ent.com/118937011/235368904-ed8476b9-0b0e-4728-9512-9374e829e683.png)

**Comment by CACG1022** [comment]
IMG 1:
![help1](https://user-images.githubusercontent.com/118937011/235369434-61be76cc-8d4f-430d-b40a-6f1bd2bcd946.png)

IMG 2:

![help2](https://user-images.githubusercontent.com/118937011/235369466-f7fff9f7-5b3c-4b4f-a21a-2a92f5e0a6e5.png)

IMG 3:
![help3](https://user-images.githubusercontent.com/118937011/235369475-7c18bd60-a251-458d-bea1-ea7d1be6c4e5.png)

PROJECT:

[Archivos.zip](https://github.com/opengisch/QField/files/11362007/Archivos.zip)

**Comment by nirvn** [comment]
@CACG1022 , alright, there are quite a few issues in your project. The most important ones are:
- you have a multitude of fields that have all the same alias (namely POLIGONO), which means when you are calling the attribute(@parent, 'POLIGONO'), you are crossing wires, _a lot_
- your CONSECUTIV field should have [x] apply default value on update checked, otherwise it'll only run the expression once upon feature creation (which can cause blanks)

When you are dissecting problems, make sure your project actually works in QGIS before you try it in QField. Your Archivos.zip didn't behave on latest QGIS 3.30, so it wouldn't work in QField.

**Comment by CACG1022** [comment]
hello @nirvn The inconvenience that it indicates could not be the real problem since if it is the only field with the alias POLIGON, additionally in the expression the layer and the field that is required to count are being specified and additionally I wanted to consult you because in previous versions like 2.7 .1 if it works correctly?. Regarding what it indicates to mark the value as default, I cannot do it because when a marker is modified, the information will be wrong since it is counting the existing markers in the current polygon. to rule it out I went back to do a new project in qgis 3.30 and the error persists 
I attach a new project made with qgis 3.30:

When placing the consecutive numbering in the polygon in QGIS it works fine, but in Qfield it doesn't work I relate video of tests:

**PROJECT:**

[MAPP8.zip](https://github.com/opengisch/QField/files/11365995/MAPP8.zip)


**test in qfield 2.7.6:**


https://user-images.githubusercontent.com/118937011/235469664-1fbaf453-f550-49b6-b4be-36f4dc6e4a8e.mp4

**test in Qgis 3.30**

https://user-images.githubusercontent.com/118937011/235470923-bb3e438b-26df-4479-9d4a-07487a7b8ef6.mp4

**Comment by JaffaSweet** [comment]
wow thanks very good
